tests/unit/core/test_withExperiment.js

```diff
@@ -229,4 +303,48 @@ describe(__filename, () => {
     const root = render({ props: { _config } });
     expect(root).toHaveProp('isExperimentEnabled', false);
   });
+
+  describe('getVariant', () => {
+    const variants = [
+      { id: 'some-variant-a', percentage: 0.1 },
+      { id: 'some-variant-b', percentage: 0.2 },
+      { id: NOT_IN_EXPERIMENT, percentage: 0.7 },
+    ];
+
+    it.each([[0, 0.01], [0, 0.1], [1, 0.11], [1, 0.3], [2, 0.31], [2, 1]])(
```

Comment:
I think `Math.random` cannot return `1`.

Reply:
It'd be nice if we had the name of the variant instead of the 0-based index in the `.each` array. That would be more readable.